File: v2/services/api/app/routers/security.py
from __future__ import annotations

from fastapi import APIRouter, Query

from app.schemas.contracts import (
    AuditListResponse,
    CredentialListResponse,
    CredentialSecretResponse,
    CredentialSummary,
    CredentialsCreateRequest,
)
from app.services.audit_service import list_audit_records
from app.services.credential_service import create_credential, get_credential_secret, list_credentials

router = APIRouter(prefix="/security", tags=["security"])


@router.post("/credentials", response_model=CredentialSummary)
def create_credential_route(payload: CredentialsCreateRequest) -> CredentialSummary:
    return create_credential(payload)


@router.get("/credentials", response_model=CredentialListResponse)
def list_credentials_route(
    limit: int = Query(default=100, ge=1, le=500),
    offset: int = Query(default=0, ge=0),
) -> CredentialListResponse:
    return list_credentials(limit=limit, offset=offset)


@router.get("/credentials/{credential_id}/secret", response_model=CredentialSecretResponse)
def get_credential_secret_route(credential_id: str) -> CredentialSecretResponse:
    return get_credential_secret(credential_id)


@router.get("/audit", response_model=AuditListResponse)
def list_audit_records_route(
    limit: int = Query(default=100, ge=1, le=500),
    offset: int = Query(default=0, ge=0),
) -> AuditListResponse:
    return list_audit_records(limit=limit, offset=offset)

File: v2/apps/designer/src/core/api/security.ts
import type {
  AuditListResponse,
  CredentialListResponse,
  CredentialSecretResponse,
  CredentialSummary
} from "../../shared/types/security";
import { apiGet, apiPost } from "./client";

export function createCredentialRequest(payload: {
  name: string;
  value: string;
  description?: string;
}) {
  return apiPost<{ name: string; value: string; description?: string }, CredentialSummary>(
    "/security/credentials",
    payload
  );
}

export function listCredentialsRequest(query: { limit?: number; offset?: number } = {}) {
  const params = new URLSearchParams();
  if (typeof query.limit === "number") {
    params.set("limit", String(query.limit));
  }
  if (typeof query.offset === "number") {
    params.set("offset", String(query.offset));
  }
  const suffix = params.toString();
  return apiGet<CredentialListResponse>(`/security/credentials${suffix ? `?${suffix}` : ""}`);
}

export function getCredentialSecretRequest(credentialId: string) {
  return apiGet<CredentialSecretResponse>(`/security/credentials/${credentialId}/secret`);
}

export function listAuditRecordsRequest(query: { limit?: number; offset?: number } = {}) {
  const params = new URLSearchParams();
  if (typeof query.limit === "number") {
    params.set("limit", String(query.limit));
  }
  if (typeof query.offset === "number") {
    params.set("offset", String(query.offset));
  }
  const suffix = params.toString();
  return apiGet<AuditListResponse>(`/security/audit${suffix ? `?${suffix}` : ""}`);
}
